src/mame/heathkit/h89.cpp

```diff
@@ -163,23 +297,118 @@ static INPUT_PORTS_START( h89 )
 INPUT_PORTS_END
 
 
+void h89_state::machine_start()
+{
+	save_item(NAME(m_gpp));
+	save_item(NAME(m_rom_enabled));
+	save_item(NAME(m_timer_intr_enabled));
+	save_item(NAME(m_floppy_ram_wp));
+
+
+	u8 *m_floppy_ram_ptr = m_floppy_ram->pointer();
+
+	// Floppy RAM - Read/Only. note: it's not rom, but behaves like it in this
+	// view.
+	m_mem_view[0].install_rom(0x1400, 0x17ff, m_floppy_ram_ptr);
+
+	// Floppy RAM - Read/write
+	m_mem_view[1].install_ram(0x1400, 0x17ff, m_floppy_ram_ptr);
+
+	// update RAM mappings based on RAM size
+	u8 *m_ram_ptr = m_ram->pointer();
+	u32 ram_size = m_ram->size();
+
+	if (ram_size == 0x10000)
+	{
+		// system has a full 64k
+		// All views will have RAM from 8k -> 64k.
+		m_mem_view[0].install_ram(0x2000, 0xffff, m_ram_ptr);
+		m_mem_view[1].install_ram(0x2000, 0xffff, m_ram_ptr);
+		m_mem_view[2].install_ram(0x2000, 0xffff, m_ram_ptr);
```

Comment:
You can install this directly into the address space: `m_maincpu->space(AS_PROGRAM).install_ram(0x2000, 0xffff, m_ram_ptr)`.  There’s no need to put it in the view when it doesn’t change.

Reply:
As mentioned above, when less than 64k, the top 8k of RAM moves between this range and the 0- 8k range.

Reply:
Yes, but because it doesn’t change dynamically, there’s no need to install into the view.  It’s only set up here on start.  In general, if you’re installing the same thing into all the view entries, it likely doesn’t need to be in the view at all.

Reply:
Only when there is 64k of memory does it not change dynamically, but it does for 16k, 32k, or 48k. Not sure how I can have common code to handle both cases without using the memory view.